### ISSUE TITLE
stree hangs out with 100% CPU

### ISSUE THREAD

**Issue by xfalcox** [ruby-syntax-tree/syntax_tree#111]
Reproduction steps:

```
git clone https://github.com/discourse/discourse-chat.git
cd discourse-chat
bundle install
bundle exec stree check --plugins=plugin/single_quotes db/migrate/*.rb
```

We are evaluating adoption syntax_tree in discourse and wanted to start with this plugin so we can get CI and bugs ironed out.

It works fine on 2.7.1 but fails on 2.8.0 so I suspect it's the parallel code.

**Comment by kddnewton** [comment]
Thanks! I'll take a look!

tbh I imagine it's not actually hanging but just taking a while. The issue is I removed the output that used to give progress feedback when I made everything parallel because it was hard to get it to sync properly without removing the benefit of the parallelization. 

I'll take a look either way.

**Comment by kddnewton** [comment]
Ahh no you're totally right something is borked with the parallel code.

**Comment by xfalcox** [comment]
Yeah, it runs very fast using 2.7.1 on the same machine :P

**Comment by kddnewton** [comment]
Oh man... it was so dumb. I was using `||` on the booleans instead of `|` so it was just skipping some threads finishing. Ugh. Pushing a fix now.

**Comment by kddnewton** [comment]
@xfalcox released in v3.0.1. Please let me know if there's anything else you run into or have questions about.
